src/typed-error.ts

```diff
@@ -19,6 +19,7 @@ export class TypedError extends Error {
 	public name: string;
 	public message: string;
 	public stack: string = '';
+	public payload: any | null;
```

Comment:
You don't need the `| null` here - firstly because it'd normally be `undefined`, but more practically because `any` includes `null` (and `undefined`) already.

I think really we're looking for this to be either undefined, or an object with any properties, right? I'd try:

```ts
{ [k: string]: any } | undefined
```